File: lat2/3.py
import pandas as pd
import mysql.connector
import matplotlib.pyplot as plt

connection = mysql.connector.connect(host='localhost', user='root', password='', database='pembelian_shopee')
query1 = "SELECT*FROM order_detail"
query2 = "SELECT*FROM sku_detail"
df1 = pd.read_sql(query1, connection)
df2 = pd.read_sql(query2, connection)
connection.close()

df = pd.merge(df1, df2, left_on='sku_id', right_on='id', how='inner')
df = df[df['is_valid'] == 1]
df['order_date'] = pd.to_datetime(df['order_date'])
df['profit'] = df['after_discount'] - (df['cogs']*df['qty_ordered'])
print(df)

tabel2021 = df[df['order_date'].dt.year == 2021]
tabel2021 = tabel2021.groupby('category')['profit'].sum().round().reset_index(name='2021')
print(tabel2021)
tabel2022 = df[df['order_date'].dt.year == 2022]
tabel2022 = tabel2022.groupby('category')['profit'].sum().round().reset_index(name='2022')
print(tabel2022)
tabelakhir = pd.merge(tabel2021, tabel2022, on='category', how='inner')
tabelakhir['%'] = ((tabelakhir['2022'] - tabelakhir['2021'])/tabelakhir['2021'])*100
tabelakhir = tabelakhir.sort_values("%", ascending=False).reset_index(drop=True)
print(tabelakhir)

File: lat2/3.r
library(RMySQL)
library(dplyr)
library(lubridate)

# Ganti 'dbname', 'username', 'password', dan 'host' sesuai dengan informasi database Anda
con <- dbConnect(MySQL(), dbname = "pembelian_shopee", user = "root", password = "", host = "localhost")

query1 <- "SELECT * FROM order_detail"
df1 <- dbGetQuery(con, query1)
query2 <- "SELECT * FROM sku_detail"
df2 <- dbGetQuery(con, query2)
dbDisconnect(con)

df <- inner_join(df1, df2, by = c('sku_id' = 'id'))
print(head(df))

df <- filter(df, is_valid == 1)
df$order_date <- as.Date(df$order_date)
df$profit <- df$after_discount - (df$cogs * df$qty_ordered)

tabel2021 <- filter(df, year(order_date) == 2021)
tabel2021 <- tabel2021 %>% group_by(category) %>% summarize(`2021` = sum(profit))

tabel2022 <- filter(df, year(order_date) == 2022)
tabel2022 <- tabel2022 %>% group_by(category) %>% summarize(`2022` = sum(profit))

tabel <- merge(tabel2021, tabel2022, by = 'category')
tabel$persentase <- (tabel$`2022` - tabel$`2021`) / tabel$`2021` *100
tabel <- arrange(tabel, desc(persentase))
print(tabel)